### PR TITLE
#81 sending calendar invitation via email

### PR DESCRIPTION
**Pull request by Evgentik**

### CODE REVIEW

File: Recrutify/Recrutify.Services/Services/SendEmailQueueService.cs [hunk 1]
```diff
@@ -43,5 +44,15 @@ public void SendEmailQueueForTest(IEnumerable<CandidateDTO> candidates, ProjectD
                 BackgroundJob.Enqueue(() => _sendEmailService.SendEmailAsync(emailRequest));
             }
         }
+
+        public void SendEmailQueueForInvite(IEnumerable<Interview> interviews)
+        {
+            var requests = _formEmailService.GetEmailRequestsForInterviewInvite(interviews);
+
+            foreach (var emailRequest in requests)
```
Comment (elizavetapavliv) [inline Recrutify/Recrutify.Services/Services/SendEmailQueueService.cs:52]: Why not to put foreach from all methods to private, not to duplicate ?

File: Recrutify/Recrutify.Services/Services/FormEmailService.cs [hunk 2]
```diff
@@ -90,5 +119,37 @@ private Generator CreateGenerator(string templatePath)
             var compiler = new HtmlFormatCompiler();
             return compiler.Compile(mailText);
         }
+
+        private StringBuilder CreateInviteFile(DateTime dateTimeInterview, string description)
+        {
+            StringBuilder str = new StringBuilder();
```
Comment (elizavetapavliv) [inline Recrutify/Recrutify.Services/Services/FormEmailService.cs:125]: var

File: Recrutify/Recrutify.Host/EmailTemplates/Interview_Email.html [hunk 3]
```diff
@@ -62,7 +62,7 @@
     <div class="wrapper">
```
Comment (elizavetapavliv) [inline Recrutify/Recrutify.Host/EmailTemplates/Interview_Email.html:62]: Rename file to Candidate_Interview_Email

File: Recrutify/Recrutify.Host/Controllers/SchedulesController.cs [hunk 4]
```diff
@@ -15,10 +14,12 @@ namespace Recrutify.Host.Controllers
     public class SchedulesController : ControllerBase
     {
         private readonly IScheduleService _scheduleService;
+        private readonly ISendEmailQueueService _sendEmailQueueService;
```
Comment (elizavetapavliv) [inline Recrutify/Recrutify.Host/Controllers/SchedulesController.cs:17]: Do you use it here?

File: Recrutify/Recrutify.Services/EmailModels/EmailRequest.cs [hunk 5]
```diff
@@ -1,11 +1,15 @@
...
     public class EmailRequest
     {
         public string ToEmail { get; set; }
 
         public string Subject { get; set; }
 
-        public string Body { get; set; }
+        public string HtmlBody { get; set; }
+
+        public StringBuilder FileBody { get; set; }
```
Comment (elizavetapavliv) [inline Recrutify/Recrutify.Services/EmailModels/EmailRequest.cs:13]: May be AttachmentBody, and not string builder - string

File: Recrutify/Recrutify.Services/EmailModels/InterviewType.cs [hunk 6]
```diff
@@ -0,0 +1,15 @@
+﻿using System.ComponentModel;
+using System.ComponentModel.DataAnnotations;
+
+namespace Recrutify.Services.EmailModels
+{
+    public enum InterviewType
+    {
+        [Description("recruity interview")]
+        RecruityInterview,
+        [Description("first technical interview")]
```
Comment (elizavetapavliv) [inline Recrutify/Recrutify.Services/EmailModels/InterviewType.cs:10]: Not first and second - entry, exit

File: Recrutify/Recrutify.Services/Events/Abstract/IUpdateStatusEventPublisher.cs [hunk 7]
```diff
@@ -5,8 +5,12 @@ namespace Recrutify.Services.Events.Abstract
 {
     public interface IUpdateStatusEventPublisher
     {
-        event Func<UpdateStatusEventArgs, Task> StatusCompleted;
+        public event Func<UpdateStatusEventArgs, Task> StatusCompleted;
+
+        public event Action<AssignedInterviewEventArgs> AssignedInterview;
```
Comment (elizavetapavliv) [inline Recrutify/Recrutify.Services/Events/Abstract/IUpdateStatusEventPublisher.cs:10]: InterviewAssigned

File: Recrutify/Recrutify.Services/Events/Abstract/IUpdateStatusEventPublisher.cs [hunk 8]
```diff
@@ -5,8 +5,12 @@ namespace Recrutify.Services.Events.Abstract
 {
     public interface IUpdateStatusEventPublisher
     {
-        event Func<UpdateStatusEventArgs, Task> StatusCompleted;
+        public event Func<UpdateStatusEventArgs, Task> StatusCompleted;
```
Comment (elizavetapavliv) [inline Recrutify/Recrutify.Services/Events/Abstract/IUpdateStatusEventPublisher.cs:8]: In interface we don't write access modifiers, remove public words

File: Recrutify/Recrutify.Services/EmailModels/EmailRequestForInvite.cs [hunk 9]
```diff
@@ -0,0 +1,15 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Recrutify.Services.EmailModels
+{
+    public class EmailRequestForInvite : EmailRequest
```
Comment (elizavetapavliv) [inline Recrutify/Recrutify.Services/EmailModels/EmailRequestForInvite.cs:9]: Remove

File: Recrutify/Recrutify.Services/Services/FormEmailService.cs [hunk 10]
```diff
@@ -60,25 +62,52 @@ public IEnumerable<EmailRequest> GetEmailRequestsForSendTest(IEnumerable<Candida
...
+                                                        name = x.Candidate.Name,
+                                                        interviewerType = x.InterviewType.GetDescription(),
+                                                        dateTime = x.AppointDateTimeUtc.AddHours(3).ToString(),
+                                                    })))
+                .Union(
+                interviews.Select(x => CreateEmailRequest(
+                                                x.User.Email,
+                                                CreateInviteFile(
+                                                    x.AppointDateTimeUtc,
+                                                    $"Candidate: {x.Candidate.Name}" +
```
Comment (elizavetapavliv) [inline Recrutify/Recrutify.Services/Services/FormEmailService.cs:87]: Put forming description in separate method

File: Recrutify/Recrutify.Services/Services/FormEmailService.cs [hunk 11]
```diff
@@ -60,25 +62,52 @@ public IEnumerable<EmailRequest> GetEmailRequestsForSendTest(IEnumerable<Candida
...
+                                                        dateTime = x.AppointDateTimeUtc.AddHours(3).ToString(),
+                                                    }))));
+        }
+
+        private EmailRequest CreateEmailRequest(string toEmail, StringBuilder fileBody, string htmlBody)
+        {
+            return new EmailRequest()
+            {
+                ToEmail = toEmail,
+                Subject = "Interview",
```
Comment (elizavetapavliv) [inline Recrutify/Recrutify.Services/Services/FormEmailService.cs:107]: Constant

File: Recrutify/Recrutify.Services/Services/FormEmailService.cs [hunk 12]
```diff
@@ -90,5 +119,37 @@ private Generator CreateGenerator(string templatePath)
...
+
+        private StringBuilder CreateInviteFile(DateTime dateTimeInterview, string description)
+        {
+            StringBuilder str = new StringBuilder();
+            str.AppendLine("BEGIN:VCALENDAR");
+
+            str.AppendLine($"PRODID: {Constants.Company.Name}");
+            str.AppendLine("VERSION:2.0");
+            str.AppendLine("METHOD:REQUEST");
+            str.AppendLine("TZOFFSETTO:+0300");
```
Comment (elizavetapavliv) [inline Recrutify/Recrutify.Services/Services/FormEmailService.cs:131]: Move + 3 in constant for all places you use it, what if next requirement will be using African time - it S not comfortable to change time in 10 places in code

File: Recrutify/Recrutify.Services/Services/SendEmailQueueService.cs (hunk 1 above)
Comment (Evgentik) [reply Recrutify/Recrutify.Services/Services/SendEmailQueueService.cs:52]: +

File: Recrutify/Recrutify.Services/Services/FormEmailService.cs (hunk 2 above)
Comment (Evgentik) [reply Recrutify/Recrutify.Services/Services/FormEmailService.cs:125]: +

File: Recrutify/Recrutify.Host/EmailTemplates/Interview_Email.html (hunk 3 above)
Comment (Evgentik) [reply Recrutify/Recrutify.Host/EmailTemplates/Interview_Email.html:62]: +

File: Recrutify/Recrutify.Services/EmailModels/EmailRequest.cs (hunk 5 above)
Comment (Evgentik) [reply Recrutify/Recrutify.Services/EmailModels/EmailRequest.cs:13]: +

File: Recrutify/Recrutify.Host/Controllers/SchedulesController.cs (hunk 4 above)
Comment (Evgentik) [reply Recrutify/Recrutify.Host/Controllers/SchedulesController.cs:17]: +

File: Recrutify/Recrutify.Services/Events/Abstract/IUpdateStatusEventPublisher.cs (hunk 7 above)
Comment (Evgentik) [reply Recrutify/Recrutify.Services/Events/Abstract/IUpdateStatusEventPublisher.cs:10]: +

File: Recrutify/Recrutify.Services/Events/Abstract/IUpdateStatusEventPublisher.cs (hunk 8 above)
Comment (Evgentik) [reply Recrutify/Recrutify.Services/Events/Abstract/IUpdateStatusEventPublisher.cs:8]: +

File: Recrutify/Recrutify.Services/EmailModels/EmailRequestForInvite.cs (hunk 9 above)
Comment (Evgentik) [reply Recrutify/Recrutify.Services/EmailModels/EmailRequestForInvite.cs:9]: +

File: Recrutify/Recrutify.Services/Services/FormEmailService.cs (hunk 10 above)
Comment (Evgentik) [reply Recrutify/Recrutify.Services/Services/FormEmailService.cs:87]: +

File: Recrutify/Recrutify.Services/Services/FormEmailService.cs (hunk 12 above)
Comment (Evgentik) [reply Recrutify/Recrutify.Services/Services/FormEmailService.cs:131]: +

File: Recrutify/Recrutify.Services/Services/FormEmailService.cs (hunk 11 above)
Comment (Evgentik) [reply Recrutify/Recrutify.Services/Services/FormEmailService.cs:107]: +

File: Recrutify/Recrutify.Host/Controllers/SchedulesController.cs [hunk 13]
```diff
@@ -1,11 +1,11 @@
 ﻿using System;
 using System.Collections.Generic;
 using System.ComponentModel.DataAnnotations;
-using System.Linq;
 using System.Threading.Tasks;
 using Microsoft.AspNetCore.Mvc;
-using Recrutify.DataAccess.Models;
 using Recrutify.Services.DTOs;
+
```
Comment (elizavetapavliv) [inline Recrutify/Recrutify.Host/Controllers/SchedulesController.cs:7]: What is it? Do you need empty line and next using after it ?

Comment (Evgentik) [reply Recrutify/Recrutify.Host/Controllers/SchedulesController.cs:7]: +

**Review by elizavetapavliv — APPROVED** [review]

**Pull request merged**